cognite/client/utils/_identifier.py

```diff
@@ -54,11 +62,21 @@ def as_dict(self, camel_case: bool = True) -> Dict[str, T_ID]:
     def as_tuple(self, camel_case: bool = True) -> Tuple[str, T_ID]:
         return self.name(camel_case), self.__value
 
-    def __str__(self) -> str:
-        return repr(self)
 
-    def __repr__(self) -> str:
-        return f"{type(self).__name__}({self.__value!r})"
+class DataModelIdentifier(IdentifierCore):
```

Comment:
A DataModel is a specific resource type in the Data Modeling API, so let's be very specific to avoid semantic confusion.
```suggestion
class DataModelingIdentifier(IdentifierCore):
```